kdump/include/runtest.sh

```diff
@@ -511,7 +506,82 @@ AppendConfig()
         shift
     done
 
-    rstrnt-report-log -l "${KDUMP_CONFIG}"
+    RhtsSubmit "${KDUMP_CONFIG}"
+}
+
+
+AppendSysconfig()
+{
+    Log "- Modifying /etc/sysconfig/kdump"
+
+    local KEY=$1
+    local ACTION=$2
+    local VALUE1=$3
+    local VALUE2=$4
+
+    if [ -z "$KEY" ] || [ -z "$ACTION" ] || [ -z "$VALUE1" ]; then
+        Error "- Missing KEY or ACTION or VALUE1."
+        return 1
+    elif [ "$ACTION" = "replace" ] && [ -z "$VALUE2" ]; then
+        Error "- Missing new_value for replacing."
+        return 1
+    elif ! grep "^$KEY=\"" "${KDUMP_SYS_CONFIG}"; then
+        Error "- Invalid KEY: $KEY."
+        return 1
+    fi
+
+    local kdump_sys_config_tmp="${KDUMP_SYS_CONFIG}.tmp"
+    \cp "${KDUMP_SYS_CONFIG}" "${kdump_sys_config_tmp}"
+
+    # Note: When KEY is "KDUMP_COMMANDLINE", ACITION add/remove/replace is actually
+    # manipulating the value of current kernel cmdline and assign
+    # it to KDUMP_COMMANDLINE.
+    # if there is no value set to KDUMP_COMMANLINE, assign current kernel cmdline
+    # to it for later string manipluation.
+    if [ "$KEY" == "KDUMP_COMMANDLINE" ] && \
+        grep "^$KEY=\"[[:space:]]*\"" "${kdump_sys_config_tmp}"; then
+
+        sed -i "/^KDUMP_COMMANDLINE=\"/d" "${kdump_sys_config_tmp}"
+        echo "KDUMP_COMMANDLINE=\"$(cat /proc/cmdline)\"" >> "${kdump_sys_config_tmp}"
+    fi
+
+    case $ACTION in
+        add)
+            # Check if the key already has a value on it, then append " $VALUE1" to the
+            # original value.
+            if grep "^$KEY=\"[[:space:]]*\"" "${kdump_sys_config_tmp}"; then
+                sed -i /^"$KEY="/d "${kdump_sys_config_tmp}"
+                echo "$KEY=\"$VALUE1\"" >> "${kdump_sys_config_tmp}"
+            else
+                sed -i  /^"$KEY="/s/\"/" $VALUE1"\"/2 "${kdump_sys_config_tmp}"
+            fi
+            ;;
+        remove)
+            sed -i  /^"$KEY="/s/"$VALUE1"//g "${kdump_sys_config_tmp}"
+            ;;
+        replace)
+            sed -i  /^"$KEY="/s/"$VALUE1"/"$VALUE2"/g "${kdump_sys_config_tmp}"
+            ;;
+        override)
+            sed -i "/^$KEY=\"/d" "${kdump_sys_config_tmp}"
+            echo "$KEY=\"$VALUE1\"" >> "${kdump_sys_config_tmp}"
+            ;;
+        *)
+            Error "Invalid action '${ACTION}' for editing kdump sysconfig."
+            false
+            ;;
+    esac
+
+    [ $? -ne 0 ] && {
+        Error "- Failed to edit kdump sysconfig"
+        rm -f "${kdump_sys_config_tmp}"
+        return 1
+    }
+
+    \mv "${kdump_sys_config_tmp}" "${KDUMP_SYS_CONFIG}"
+    RhtsSubmit "${KDUMP_SYS_CONFIG}"
```

Comment:
Same comment here about the `Rhts*` -> restraint

Reply:
The rhts-* can be moved away all together in the furture within RhtsSubmit wrapper.